From 2c23434a9fcf513ff4b76d8bfe62272b0454d9f1 Mon Sep 17 00:00:00 2001
From: Stephan Gerhold <stephan@gerhold.net>
Date: Sat, 30 Nov 2019 17:41:46 +0100
Subject: [PATCH 101/210] arm64: dts: qcom: msm8916-samsung-serranove: Add
 sound and modem (v7)

SOUND:
v2: Rename dai-link nodes, add labels

MODEM:
v2: Set lpass@7708000 to status = "disabled"
v3: add q6asm (MultiMedia Playback/Capture) routes
v4: Add CS-Voice link, add MultiMedia4 for compressed playback
v5: Setup memshare / GPS memory
v6: wcnss/venus no longer need to be relocated

v7: Combine sound and modem commits and drop extra -modem.dts
    The no-modem variant is now patched dynamically by lk2nd instead.

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
---
 .../dts/qcom/msm8916-samsung-serranove.dts    | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-samsung-serranove.dts b/arch/arm64/boot/dts/qcom/msm8916-samsung-serranove.dts
index 58dfbfff4c7..5f35b3e2a11 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-samsung-serranove.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-samsung-serranove.dts
@@ -4,6 +4,8 @@
 /dts-v1/;
 
 #include "msm8916-pm8916.dtsi"
+#include "msm8916-modem.dtsi"
+
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/interrupt-controller/irq.h>
@@ -39,6 +41,16 @@ tz-apps@85500000 {
 			reg = <0x0 0x85500000 0x0 0xb00000>;
 			no-map;
 		};
+
+		mpss_mem: mpss@86800000 {
+			reg = <0x0 0x86800000 0x0 0x5a00000>;
+			no-map;
+		};
+
+		gps_mem: gps@8c200000 {
+			reg = <0x0 0x8c200000 0x0 0x200000>;
+			no-map;
+		};
 	};
 
 	gpio-keys {
@@ -300,6 +312,14 @@ &sdhc_2 {
 	no-1-8-v;
 };
 
+&sound {
+	status = "okay";
+	audio-routing =
+		"AMIC1", "MIC BIAS External1",
+		"AMIC2", "MIC BIAS Internal2",
+		"AMIC3", "MIC BIAS External1";
+};
+
 &usb {
 	status = "okay";
 	extcon = <&muic>, <&muic>;
-- 
2.34.1

